benchmark/utils/general.hpp

```diff
@@ -392,6 +392,21 @@ gko::remove_complex<ValueType> compute_norm2(const vec<ValueType> *b)
 }
 
 
+template <typename ValueType>
+gko::remove_complex<ValueType> compute_direct_error(const gko::LinOp *solver,
+                                                    const vec<ValueType> *b,
+                                                    const vec<ValueType> *x)
+{
+    auto exec = solver->get_executor();
+    auto ref_solver = gko::clone(gko::ReferenceExecutor::create(), solver);
+    auto one = gko::initialize<vec<ValueType>>({1.0}, exec);
+    auto neg_one = gko::initialize<vec<ValueType>>({-1.0}, exec);
+    auto res = clone(x);
+    ref_solver->apply(lend(one), lend(b), lend(neg_one), lend(res));
+    return compute_norm2(lend(res));
```

Comment:
I think you should clone `res` also to be on reference executor ? Otherwise the norm2 gets computed on the main executor and might have differences due to partial reduction orderings ?